tuf/store/httpstore.go

```diff
@@ -270,13 +275,54 @@ func (s HTTPStore) buildURL(uri string) (*url.URL, error) {
 	return s.baseURL.ResolveReference(sub), nil
 }
 
-// GetKey retrieves a public key from the remote server
-func (s HTTPStore) GetKey(role string) ([]byte, error) {
+// GetKey retrieves the most recently created (whether it is signed in yet or not)
+// public key for the given role from the remote server
+func (s HTTPStore) GetKey(role string) (data.PublicKey, error) {
+	return s.requestKey(role, nil)
+}
+
+// RotateKey rotates a key on the remote server and returns the new public key.  This requires
+// a request with a short expiry time (to limit replay), signed by at least one root key.
+// Signing with the root key proves that the client making the request has the capability of
+// actually rotating the key, and is not just making a spurious rotation request.
+func (s HTTPStore) RotateKey(role string, cs signed.CryptoService, roots ...data.PublicKey) (data.PublicKey, error) {
+	req := &data.SignedCommon{
+		Type:    role,
+		Expires: time.Now().Add(5 * time.Minute),
+	}
+	reqJSON, err := json.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+	signedReq := &data.Signed{Signed: reqJSON}
+	if err := signed.Sign(cs, signedReq, roots...); err != nil {
```

Comment:
I guess it depends what you think of the signed request as.  If it should be part of any rotate key API, no matter if we implement a GRPC remote API, an in-memory API, etc, that all requires a signed request, and (and maybe in fact clients should never be able to rotate any key, even locally, without having the root key on hand?), then I agree it should probably go into the client.

If we are thinking of this as just something our HTTP server requires, as in it's part of the server's API and not a generic `RemoteStore` API, then it should probably live in the client for the notary server, which seems to be `HTTPStore`, since that's the thing which knows how to speak the server API.